sklearn/svm/base.py

```diff
@@ -287,11 +287,14 @@ def _sparse_fit(self, X, y, sample_weight, solver_type, kernel,
         n_SV = self.support_vectors_.shape[0]
 
         dual_coef_indices = np.tile(np.arange(n_SV), n_class)
-        dual_coef_indptr = np.arange(0, dual_coef_indices.size + 1,
-                                     dual_coef_indices.size / n_class)
-        self.dual_coef_ = sp.csr_matrix(
-            (dual_coef_data, dual_coef_indices, dual_coef_indptr),
-            (n_class, n_SV))
+        if dual_coef_indices.size == 0:
```

Comment:
I would replace this check with if `n_SV == 0` and only declare `dual_coef_indices` where it is actually used, i.e. in the `else` clause.

Reply:
replaced with `not n_SV` check. If you'd prefer `n_SV == 0` , just let me know